acceptance/internal/prepare_server.go

```diff
@@ -103,6 +109,9 @@ func startDedicatedServer(t *testing.T,
 
 	if logRequests {
 		s.ResponseCallback = func(request *testserver.Request, response *testserver.EncodedResponse) {
+			mu.Lock()
```

Comment:
nit: Not necessary, t.Logf is threadsafe. Fine to be explicit though.

Reply:
Very good point, Logf is safe according to the docs:

> https://pkg.go.dev/testing#T
> The other reporting methods, such as the variations of Log and Error, may be called simultaneously from multiple goroutines.

Reply:
Removed.